x/ccv/provider/keeper/distribution.go

```diff
@@ -57,32 +68,165 @@ func (k Keeper) GetAllConsumerRewardDenoms(ctx sdk.Context) (consumerRewardDenom
 	return consumerRewardDenoms
 }
 
-// TransferRewardsToFeeCollector transfers all consumer rewards to the fee collector address
-func (k Keeper) TransferRewardsToFeeCollector(ctx sdk.Context) {
-	// 1. Get the denom whitelist from the store
-	denoms := k.GetAllConsumerRewardDenoms(ctx)
+// AllocateTokens performs reward and fee distribution to all validators based
+// on the Partial Set Security distribution specification.
```

Comment:
maybe comment on what happens to the remaining funds that were not allocated to a validator
(from reading the code, I think they are just sent to the community pool)